scripts/gen-doc.py

```diff
@@ -185,7 +185,8 @@ def _main():
     s += _get_parameter_flags_block()
     s += _get_parameter_choices_block(data["choices"])
 
-    print(s.rstrip())
+    with open(sys.argv[2], "w") as f:
```

Comment:
Do we need to force the encoding here ?

Reply:
Might be reasonable if we update the doc from Windows. Added.

I also added more forced encoding in the first commit just in case, and in the intermediate commit that still uses stdout (`gen_doc: replace program with a Python script parsing the specs`), I did add the `PYTHONIOENCODING` thing (which later in the history goes away).